src/app/modules/item/task-communication/rxjschannel.ts

```diff
@@ -35,31 +40,70 @@ export class RxMessagingChannel {
     return this.innerChan.unbind(method, doNotPublish);
   }
 
-  bind<T>(method: string, observable?: (params: T) => Observable<unknown>, doNotPublish?: boolean): MessagingChannel {
+  /** Bind a local method, allowing the remote task to call it */
+  bind<T>(method: string, observable?: (params: T) => Observable<unknown>, validator?: D.Decoder<unknown, T>,
+    selector?: (params: any[]) => unknown, doNotPublish?: boolean): MessagingChannel {
     // Create a callback wrapping the observable bound
-    function callback(transaction: MessageTransaction, params: T): void {
+    function callback(transaction: MessageTransaction, ...params: any[]): void {
       if (!observable) {
         return;
       }
-      const cb$ = observable(params);
+      // Select params
+      const actualSelector = selector
+        ? selector
+        : (result: any[]) : unknown => (result.length > 0 ? result[0] : undefined);
```

Comment:
Note to self:
To me, selector logic should not be internalise to the bind method since
- it is not the method purpose to map anything
- it can be done easily using rxjs
- it is (currently) used only once